cmd/observe/flows_filter.go

```diff
@@ -546,6 +547,17 @@ func (of *flowFilter) set(f *filterTracker, name, val string, track bool) error
 			f.HttpUrl = append(f.GetHttpUrl(), val)
 		})
 
+	case "http-header":
+		f.apply(func(f *flowpb.FlowFilter) {
+			h := strings.Split(val, ":")
+			if len(h) > 0 {
+				hVal := h[len(h)-1]
+				key := strings.Join(h[:len(h)-1], ":")
```

Comment:
I think this should be the other way around. Everything before the first colon is the key and everything after is the value. I think `strings.SplitN` as mentioned by Anna would work as intended here. Another easier option might be [`strings.Cut`](https://pkg.go.dev/strings#Cut):

```go
key, hVal, _ := strings.Cut(h, ":")
```

Reply:
Got it! I doubled checked and only the values can contain `:`. Changed that.